nixos/modules/security/please.nix

```diff
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.security.please;
+in
+{
+  options.security.please = {
+    enable = mkEnableOption (mdDoc ''
+      please, a tool which allows non-root users to execute commands as root
+    '');
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.please;
+      defaultText = literalExpression "pkgs.please";
+      description = mdDoc ''
+        Which package to use for {command}`please`;
+      '';
+    };
+
+    wheelAsRoot = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Whether users of the `wheel` group can execute commands or edit files
+          using {command}`please` and {command}`pleaseedit` respectively.
+        '';
+      };
+
+      needsPassword = mkOption {
+        type = types.bool;
+        default = true;
+        description = lib.mdDoc ''
+          Whether users of the `wheel` group must provide a password to run
+          commands or edit files {command}`please` and {command}`pleaseedit`
+          respectively.
+        '';
+      };
+    };
+
+    settings = mkOption {
+      type = types.attrs;
+      default = { };
+      example = {
+        jim_become_root = {
+          name = "jim";
+          target = "root";
+          rule = ".*";
+          require_pass = false;
+        };
+        jim_hosts = {
+          name = "jim";
+          type = "edit";
+          target = "root";
+          rule = "/etc/hosts";
+          editmode = 644;
+          require_pass = false;
+        };
+      };
+      description = mdDoc ''
+        Please configuration. Refer to
+        <https://github.com/edneville/please/blob/master/please.ini.md> for
+        details.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    security.wrappers =
+      let
+        owner = "root";
+        group = "root";
+        setuid = true;
+      in
+      {
+        please = {
+          source = "${cfg.package}/bin/please";
+          inherit owner group setuid;
+        };
+        pleaseedit = {
+          source = "${cfg.package}/bin/pleaseedit";
+          inherit owner group setuid;
+        };
+      };
+
+    environment = {
+      systemPackages = [ cfg.package ];
+
+      etc."please.ini" = {
```

Comment:
At some point, we'll also want to check the validity of this file at build time, like I did with `doas` (maybe `sudo` does the same, but I don't play with that module much). However, this is currently blocked on https://github.com/edneville/please/issues/4 (it's not currently possible to check the validity of the config file without `please` being setuid).